Support tailing from a seekable reader

Mook-as opened the pull request:
This adds the possibility to use tail on a `io.ReadCloser` that's also a `io.Seeker`.  It mostly does nothing very interesting in that case, but is easier for consumers to deal with.

sschepens commented:
I dont know much, but isn't it better to create an interface which is a ReadCloser+Seeker and use that instead of casting to io.Seeker to see if the variable implements the interface? this way we get compile time guarantees and not runtime

Mook-as commented:
Sorry about the late reply! (Vacation, and other things)

I thought when I originally made the change that it wouldn't work for whatever reason (the interface not being available and whatnot), but on testing it seems to work, so it's now fixed.  Thanks!

Addressed in 1b19cf2328f0061696330f81987fddf1cc378a26

Nino-K commented:
@Mook-as This PR can not be merged at this time, as the following [condition](https://github.com/hpcloud/tail/blob/master/tail.go#L222-L227) works against your changes. 
The best solution would be just point the BYOD related work to a fork and we can revisit this at a later point, as there is not an easy, viable solution at this point.
